src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -108,25 +142,89 @@ private async Task<IEnumerable<RazorCodeAction>> GetCodeActionsAsync(CodeActionP
                 return null;
             }
 
+            var sourceText = await documentSnapshot.GetTextAsync().ConfigureAwait(false);
+
+            var context = new RazorCodeActionContext(
+                request,
+                documentSnapshot,
+                codeDocument,
+                sourceText,
+                _languageServerFeatureOptions.SupportsFileManipulation);
+
+            return context;
+        }
+
+        private async Task<IEnumerable<RazorCodeAction>> GetCSharpCodeActionsAsync(RazorCodeActionContext context, CancellationToken cancellationToken)
+        {
+            var csharpCodeActions = await GetCSharpCodeActionsFromLanguageServerAsync(context, cancellationToken);
+            var filteredCSharpCodeActions = await FilterCSharpCodeActionsAsync(context, csharpCodeActions, cancellationToken);
+
+            return filteredCSharpCodeActions;
+        }
+
+        private async Task<IEnumerable<RazorCodeAction>> FilterCSharpCodeActionsAsync(RazorCodeActionContext context, IEnumerable<RazorCodeAction> csharpCodeActions, CancellationToken cancellationToken)
```

Comment:
Is it filter or is it "remap"?

Reply:
Filter b/c we take the full set of C# code actions supplied by Roslyn, and filter out the ones we support in Razor.